src/parser/csql_grammar.y

```diff
@@ -5185,45 +5192,125 @@ only_all_class_spec
 		DBG_PRINT}}
 	;
 
-class_name
-	: identifier DOT identifier
+user_specified_name_without_dot
+	: identifier_without_dot DOT identifier_without_dot
 		{{
 
-			PT_NODE *user_node = $1;
-			PT_NODE *name_node = $3;
+			PT_NODE *user = $1;
+			PT_NODE *name = $3;
 
-			if (name_node != NULL && user_node != NULL)
+			if (user
+			    && name
+			    && user->node_type == PT_NAME
+			    && name->node_type == PT_NAME)
```

Comment:
identifier_without_dot is always PT_NAME.

Reply:
Okay. I'll just check for NULL.